### PR TITLE
doc/config-v3_0: Add notes about uniqueness

### PR DESCRIPTION
**Pull request by ajeddeloh**
Add notes specifying which fields must be unique.

Will add another commit for 3.1.0-experimental once this one looks good.

### CODE REVIEW

File: doc/configuration-v3_0.md [hunk 1]
```diff
@@ -18,15 +18,15 @@ The Ignition configuration is a JSON document conforming to the following specif
...
         * **_verification_** (object): options related to the verification of the certificate.
           * **_hash_** (string): the hash of the certificate, in the form `<type>-<value>` where type is sha512.
 * **_storage_** (object): describes the desired state of the system's storage devices.
-  * **_disks_** (list of objects): the list of disks to be configured and their options.
+  * **_disks_** (list of objects): the list of disks to be configured and their options. Every entry must have a unique `device`.
     * **device** (string): the absolute path to the device. Devices are typically referenced by the `/dev/disk/by-*` symlinks.
     * **_wipeTable_** (boolean): whether or not the partition tables shall be wiped. When true, the partition tables are erased before any further manipulation. Otherwise, the existing entries are left intact.
-    * **_partitions_** (list of objects): the list of partitions and their configuration for this particular disk.
+    * **_partitions_** (list of objects): the list of partitions and their configuration for this particular disk. Every partition must have a unique `number`, or if 0 is specified, a unique `label.
```
Comment (arithx) [inline doc/configuration-v3_0.md]: missing backtick on `label`

File: doc/configuration-v3_0.md [hunk 2]
```diff
@@ -69,7 +69,7 @@ The Ignition configuration is a JSON document conforming to the following specif
     * **_group_** (object): specifies the group of the owner.
       * **_id_** (integer): the group ID of the owner.
       * **_name_** (string): the group name of the owner.
-  * **_directories_** (list of objects): the list of directories to be created.
+  * **_directories_** (list of objects): the list of directories to be created. Every file, directory, and link must have a unique path.
```
Comment (arithx) [inline doc/configuration-v3_0.md]: Files wraps `path` in back-ticks while directories & links do not. Should probably update both directories & links to wrap them (as well as `name` in systemd units, drop-ins, users, & groups) for consistency.

**Comment by ajeddeloh** [comment]
Fixed.

File: doc/configuration-v3_0.md [hunk 3]
```diff
@@ -18,15 +18,15 @@ The Ignition configuration is a JSON document conforming to the following specif
...
         * **_verification_** (object): options related to the verification of the certificate.
           * **_hash_** (string): the hash of the certificate, in the form `<type>-<value>` where type is sha512.
 * **_storage_** (object): describes the desired state of the system's storage devices.
-  * **_disks_** (list of objects): the list of disks to be configured and their options.
+  * **_disks_** (list of objects): the list of disks to be configured and their options. Every entry must have a unique `device`.
     * **device** (string): the absolute path to the device. Devices are typically referenced by the `/dev/disk/by-*` symlinks.
     * **_wipeTable_** (boolean): whether or not the partition tables shall be wiped. When true, the partition tables are erased before any further manipulation. Otherwise, the existing entries are left intact.
-    * **_partitions_** (list of objects): the list of partitions and their configuration for this particular disk.
+    * **_partitions_** (list of objects): the list of partitions and their configuration for this particular disk. Every partition must have a unique `number`, or if 0 is specified, a unique `label`.
```
Comment (arithx) [inline doc/configuration-v3_0.md]: In the `number` field we have `If zero, use the next available partition slot`, do you think it's worthwhile to write it out here as well for consistency?

Comment (ajeddeloh) [reply doc/configuration-v3_0.md]: I don't think it's necessary, but if you think it is, I can add it.

Comment (arithx) [reply doc/configuration-v3_0.md]: I don't have a particularly strong opinion on changing it, I generally prefer to keep it consistent but I could see a case for this being a representation of the actual value that causes it to switch to being keyed off of `label`.

Not a blocker imo.

Comment (ajeddeloh) [reply doc/configuration-v3_0.md]: I'll leave it for now unless someone else chimes in. The docs on `number` immediately follow it, so I'm not too worried.

**Comment by ajeddeloh** [comment]
Updated to also update the experimental spec doc

**Pull request merged**